src/core/Akka/Actor/Internal/ActorSystemImpl.cs

```diff
@@ -177,10 +177,8 @@ private void ConfigureExtensions(IEnumerable<IExtensionId> extensionIdProviders)
         public override object RegisterExtension(IExtensionId extension)
         {
             if(extension == null) return null;
-            if(!_extensions.ContainsKey(extension.ExtensionType))
-            {
-                _extensions.TryAdd(extension.ExtensionType, new Lazy<object>(() => extension.CreateExtension(this)));
-            }
```

Comment:
this can be the couse. Should that "return extension.Get(this);" not actualy using the result of _extensions.GetOrAdd ?